### ISSUE TITLE
How to handle github-style checkboxes?

### ISSUE THREAD

**Issue by hoylemd** [markdownlint/markdownlint#264]
Hi folks,

I'm using mdl with Syntastic in vim to write my markdown documents, and they often contain checklists like this:

```markdown
- [ ] first item
- [x] second item
- [ ] third item
```

which renders in github like so:

- [ ] first item
- [x] second item
- [ ] third item

but when I run mdl on a document containing this I get this error:

>> No link definition for link ID 'x' [markdown/mdl]

(it says ' ' on the lines where there isn't an 'x' in the box)

Is there a style or ruleset pre-defined to support github flavour markdown? Alternatively, is there a specific rule I can disable via my .mdlrc file? If neither of those are valid, is there some other way to convince mdl that these are checkboxes and not links?

Thanks for the help. Once we get it sorted, I'll put up a PR to add documentation on how to solve this, so you don't keep getting issues about it :)

**Comment by hoylemd** [comment]
Some additional context:

this warning seems to be coming from Kramdown, not markdownlint directly. I'm going to try to squelch the warnings somehow for now, but it might require some changes to how kramdown is used, or even kramdown itself

**Comment by hoylemd** [comment]
I figured out how to squelch the warnings via syntastic. Add this to your vimrc:

```vim
let g:syntastic_quiet_messages = { 'regex': ["No link definition for link ID ' '"] }
```

That will silence this warning in particular, which solved my problem.

**Comment by jaymzh** [comment]
Yes, I agree we should definitely handle that better. It shouldn't be too hard to treat brackets in a list, with either a single space between them or an X between them, differently. Maybe something @NaomiReeves or @bwann wanna take a wack at?

**Comment by jaymzh** [comment]
So, I spent some time digging into this today with @NaomiReeves . It's ... complicated.

As you found out, the problem here is not our rules, but the parsing engine. Kramdown is nice and modular, so there's several parsers. The default one is eponymous `Kramdown`, which is a superset of Markdown. There's also `Markdown` for strict markdown, and `GFM` for Github-flavored-markdown. The latter inherits from `Kramdown`, so it's all of those goodies, plus extra bits for GFM.

Makdownlint uses... it's own parser which inherits from `Kramdown`, but overrides one small thing that lets it use the `blocked-fences` parsing from GFM. Due some internals of the way the kramdown stuff works, that specific case happened to be easy, but in this case it's not easy to swap out the list-parsing.

So there's two options:

1. We can copy-paste the [`parse_list` method from the gfm parser](https://github.com/gettalong/kramdown/blob/43639dfddc1e9208a327f9bf7c58ace7a71cf350/lib/kramdown/parser/gfm.rb#L159) into our wrapper-parser and solve this case, but we will not get any bugfixes or enhancements to it without babysitting that code and copy-pasting it all.
2. We can try to move completely to the GFM parser. I did this on a whim and 5 tests fail, 1 of which is just an ordering-of-output red-herring, but the other 4 of which are real and affect a *lot* of rules... but it's only 4 out of 344, so it's **likely** (but not certain) we can adjust our rules to handle the changes in the parsed AST. For the curious, [here are the failures](https://gist.github.com/jaymzh/c168034be0e33d78802cd5b74d6cea93).

I'm very, very much not a fan of number 1. Number 2 is more work and isn't guaranteed to work, **however**, if it does work, we'll also get some other features people are likely to file requests about in the future such as "strike through" and such.

**Comment by hoylemd** [comment]
I agree #2 is a much better option. Thanks for looking into it @NaomiReeves

**Comment by anarcat** [comment]
> let g:syntastic_quiet_messages = { 'regex': ["No link definition for link ID ' '"] }

is there a way to do such an override in markdownlint itself?
